### PR TITLE
[Readline] Add builder

### LINKED ISSUE #227: Add a builder for GNU Readline
Looks like there's one at https://github.com/benlorenz/readlineBuilder that uses an old version of BinaryBuilder. It'd be nice to have readline in Yggdrasil.

### PR DESCRIPTION
**Pull request by giordano**
Fix #227

### CODE REVIEW

File: R/Readline/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+export CPPFLAGS="-I${prefix}/include"
+
+./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} --with-curses
+make -j${nproc} SHLIB_LIBS=-lncurses
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [p for p in supported_platforms() if !(p isa Windows)]
```
Comment (giordano) [inline R/Readline/build_tarballs.jl]: I didn't even try to build for Windows, but I think this is usually not needed for this platform?

Comment (ararslan) [reply R/Readline/build_tarballs.jl]: Might as well skip it for now, unless someone finds they need it. Lua doesn't seem to use readline on Windows, at least.

**Comment by ararslan** [comment]
What needs to happen here now that #229 has been merged?

**Comment by giordano** [comment]
Fix https://github.com/JuliaPackaging/Yggdrasil/pull/225#issuecomment-558529896 :disappointed:

**Comment by giordano** [comment]
I reverted #225 in #236.  Now we need https://github.com/JuliaRegistries/General/pull/5880 to be merged, after that I'll fix this builder and then we can go

**Comment by ararslan** [comment]
I merged the General PR.

As an aside, the commit message there has
```
Registrator tree SHA: unknown
```
in it, which makes me thing something might be amok with the recent BinaryBuilder automatic registration changes...

**Comment by giordano** [comment]
Ok, I've updated the builder and spent those 3-4 minutes to make it work for Windows, too.

> As an aside, the commit message there has

:thinking:

**Comment by ararslan** [comment]
cc @staticfloat regarding the Registrator thing. In the meantime this LGTM.

**Comment by giordano** [comment]
All [commits by jlbuild](https://github.com/JuliaRegistries/General/commits?author=jlbuild) have this, so it's not something new :grin:

**Pull request merged**